jdaviz/configs/mosviz/helper.py

```diff
@@ -288,31 +290,32 @@ def _row_click_message_handler(self, msg):
     def _handle_image_zoom(self, msg):
         mos_data = self.app.data_collection['MOS Table']
 
-        # trigger zooming the image, if there is an image
-        if mos_data.find_component_id("Images") is not None:
-            if msg.shared_image:
-                center, height = self._zoom_to_object_params(msg)
-            else:
-                try:
-                    center, height = self._zoom_to_slit_params(msg)
-                except IndexError:
-                    # If there's nothing in the spectrum2d viewer, we can't get slit info
-                    return
-
-            if center is None or height is None:
-                # Can't zoom if we couldn't figure out where to zoom (e.g. if RA/Dec not in table)
-                return
+        if mos_data.find_component_id("Images") is None:
+            return
 
-            imview = self.app.get_viewer(self._default_image_viewer_reference_name)
+        imview = self.app.get_viewer(self._default_image_viewer_reference_name)
+
+        # trigger zooming the image, if there is an image
+        if msg.shared_image:
+            center, height = self._zoom_to_object_params(msg)
+        else:
+            center, height = self._zoom_to_slit_params(msg)
 
-            image_axis_ratio = ((imview.axis_x.scale.max - imview.axis_x.scale.min) /
-                                (imview.axis_y.scale.max - imview.axis_y.scale.min))
+        if height is not None:
+            image_axis_ratio = (abs(imview.state.x_max - imview.state.x_min) /
+                                abs(imview.state.y_max - imview.state.y_min))
+            xh = image_axis_ratio * height
```

Comment:
I generally have a disdain towards two-letter vars that aren't self explanatory. Any way we could add a meaningful name to `xh`?

Reply:
Changed.